python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -414,6 +414,17 @@ def implements_accurate_memory_accounting(self) -> bool:
     def supports_fusion(self) -> bool:
         return self._supports_fusion
 
+    def num_active_tasks(self) -> int:
```

Comment:
If I'm understanding correctly, this change makes `len(get_active_tasks()) != num_active_tasks` for map operators. I think that might get confusing.

@raulchen do have any ideas for how we can address the issues while avoiding this discrepancy?

Reply:
I thought of this as well. It's indeed a bit confusing.
I also thought of separating data and metadata tasks. But that seems overkill.
I think maybe I'll just update the comments to clarify for now.